ext/native/thin3d/thin3d_vulkan.cpp

```diff
@@ -290,14 +291,20 @@ class VKTexture;
 class VKBuffer;
 class VKSamplerState;
 
+enum {
+	MAX_BOUND_TEXTURES = 2
+};
+
 struct DescriptorSetKey {
-	VkImageView imageView_;
-	VKSamplerState *sampler_;
+	VkImageView imageViews_[MAX_BOUND_TEXTURES];
+	VKSamplerState *samplers_[MAX_BOUND_TEXTURES];
 	VkBuffer buffer_;
 
 	bool operator < (const DescriptorSetKey &other) const {
-		if (imageView_ < other.imageView_) return true; else if (imageView_ > other.imageView_) return false;
-		if (sampler_ < other.sampler_) return true; else if (sampler_ > other.sampler_) return false;
+		for (int i = 0; i < MAX_BOUND_TEXTURES; ++i) {
```

Comment:
This loop seem weird to me, it have return in it so unless they are the same it will end asap, is this fine is this order? I'm not sure what is this used for tho'.

Reply:
Right, this is just a compare to say - is this key the same?  It's so it can build the map.  This should be fine.

-[Unknown]